### PR TITLE
Add camera support for windows.

### PR DESCRIPTION
**Pull request by shenweip**
Implement this with MF api,referenced some Microsoft docs and samples.
I tested **Invizimals**, it works fine on my laptop(with a Microsoft HD Webcam).
**Preview:**

Also has a basic support for multi camera switching.

EDIT by hrydgard: Helps #12337

### CODE REVIEW

**Comment by hrydgard** [comment]
Very cool! How far back in Windows versions does this API work? Is it supported on Vista? 

I was dreading doing this because of DirectShow which I still have nightmares about (only slightly kidding)...

**Comment by shenweip** [comment]
Accordings to docs, most of them are supported on vista, maybe few are not supported though,like MFEnumDeviceSources function.It has a full support on win7.I think not many users on vista?

**Comment by hrydgard** [comment]
It's just that I don't want a change to our minimum requirements to be accidental, it should be intentional. Currently PPSSPP supports Vista and forwards, so it would be good to be sure - if this API doesn't work there, we could simply disable it on Vista, keeping PPSSPP working for Vista users even though they won't get this feature.

**Comment by shenweip** [comment]
Yeah, it's right, I will comfirm this through VMware.

File: UI/NativeApp.cpp [hunk 1]
```diff
@@ -879,6 +888,11 @@ void NativeShutdownGraphics() {
 	winAudioBackend = nullptr;
 #endif
 
+#ifdef _WIN32
+	delete winCamera;
```
Comment (Florin9doi) [inline UI/NativeApp.cpp]: if (winCamera) ?

Comment (shenweip) [reply UI/NativeApp.cpp]: Right.

Comment (hrydgard) [reply UI/NativeApp.cpp]: Not really necessary as delete handles nullptr, but it's fine either way I think.

**Comment by Florin9doi** [comment]
I've tested the first commit and I've observed that the image is stretched because the games ask for 480x272, but the webcam provides 640x480 (I think) which have a different aspect ratio.
This can be fixed by passing to sws_scale/sws_getContext a smaller source height

**Comment by hrydgard** [comment]
If you pass a smaller source height you should also offset, to center it vertically, I guess. Though I'm not sure how much that matters for these games...

**Comment by hrydgard** [comment]
Improving stuff like that can be done later, good to get the functionality in, so I'm merging.

**Pull request merged**